### PR TITLE
net-p2p/i2p: bump to 0.9.28

### PR DESCRIPTION
**Pull request by tharvik**
Fix gbugs#602642.

Some small improvements
 * only show message about ecdsa if the flag is not enable and simplify it
 * remove some ecdsa dependency, as they are useless
    * if we have `dev-java/icedtea[nss,-sunec]` and `dev-java/icedtea[nss]`, only keep the latter
 * use variable in elog about configuration

### CODE REVIEW

File: net-p2p/i2p/i2p-0.9.28.ebuild [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+# Until the deps reach other arches
+KEYWORDS="~amd64 ~x86"
+IUSE="+ecdsa nls"
+
+# dev-java/ant-core is automatically added due to java-ant-2.eclass
+CDEPEND="dev-java/bcprov:1.50
```
Comment (chewi) [inline net-p2p/i2p/i2p-0.9.28.ebuild]: Rename this to `CP_DEPEND` and you can then remove `EANT_GENTOO_CLASSPATH` below.

File: net-p2p/i2p/i2p-0.9.28.ebuild [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+)
+
+pkg_setup() {
+	java-pkg-2_pkg_setup
+
+	enewgroup i2p
+	enewuser i2p -1 -1 "${I2P_CONFIG_HOME}" i2p
+}
+
+src_unpack() {
```
Comment (chewi) [inline net-p2p/i2p/i2p-0.9.28.ebuild]: This whole function can simply be replaced with `JAVA_ANT_REWRITE_CLASSPATH="true"`.

File: net-p2p/i2p/i2p-0.9.28.ebuild (hunk 1 above)
Comment (tharvik) [inline net-p2p/i2p/i2p-0.9.28.ebuild]: Done.

File: net-p2p/i2p/i2p-0.9.28.ebuild (hunk 2 above)
Comment (tharvik) [inline net-p2p/i2p/i2p-0.9.28.ebuild]: It doesn't work, warn me about much missing symbols
```
    [javac] /var/tmp/portage/net-p2p/i2p-0.9.28/work/i2p-0.9.28/router/java/src/net/i2p/data/i2np/BuildRequestRecord.java:5: error:
cannot find symbol
    [javac] import net.i2p.I2PAppContext;
    [javac]               ^
    [javac]   symbol:   class I2PAppContext
    [javac]   location: package net.i2p
    [javac] /var/tmp/portage/net-p2p/i2p-0.9.28/work/i2p-0.9.28/router/java/src/net/i2p/data/i2np/BuildRequestRecord.java:6: error:
cannot find symbol
    [javac] import net.i2p.data.Base64;
    [javac]                    ^
    [javac]   symbol:   class Base64
    [javac]   location: package net.i2p.data
```

Comment (chewi) [inline net-p2p/i2p/i2p-0.9.28.ebuild]: Weird, should work if `JAVA_ANT_REWRITE_CLASSPATH` is set. Not a big deal.

**Comment by chewi** [comment]
Please amend when making changes  for reviews to keep the commit history clean.

**Comment by tharvik** [comment]
Squashed as wanted.

**Comment by tharvik** [comment]
@gentoo-repo-qa-bot got it wrong btw, is there anything more to do or does it look good for you @chewi?

**Pull request closed without merging**